DEVELOPING.md

```diff
@@ -1,75 +1,210 @@
+
+Build & Test Toolbox IDE
+------------------------
+Install the following dependencies to your path:
+ * [Java Development Kit](https://adoptium.net/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+ * [Apache Maven](https://maven.apache.org/) version 3.9.7+
+
+Clone this repo & open a shell in its root, then run:
+```bash
+mvn install -Dmaven.test.skip=true # Builds Toolbox
+mvn verify # Runs tests
+```
 
-Firstly this is not complete. If this doesn't have information you need, please
-engage with tlaplus contributors on Github
-and add new knowledge, instructions, insights, or tips back to this document to
-help the next potential contributor
-(or even yourself a few months down the track once you've forgotten everything).
+The Toolbox distributables will be output in the `toolbox/org.lamport.tla.toolbox.product.product/target/products` directory.
 
+Developing with the Eclipse IDE
+-------------------------------
+You are free to use your own editor, but this project is largely developed in the [Eclipse](https://www.eclipse.org/) integrated development environment (IDE).
 
-Building TLC / `tlatools.jar`
-=============================
+### Developing the TLA⁺ Tools in Eclipse
 
-While the Toolbox is maintained in [Eclipse](https://github.com/tlaplus/tlaplus/tree/master/general/ide), sometimes changes to TLC do not require building the entire GUI. In such cases, it can be beneficial to compile TLC on the command line. 
+Developing the TLA⁺ Tools in Eclipse requires the [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2024-06/r/eclipse-ide-rcp-and-rap-developers) edition, or any other edition that comes with the OSGi Plugin Development Extension.
+After you have installed Eclipse, use it open the `tlatools/org.lamport.tlatools` project directory.
+You should then be able to build, run, test, and debug the TLA⁺ Tools.
 
-While `Maven` is used for the Toolbox (see below), `ant` can be used for the command line. It can be installed with `brew` or `apt` with the package name `ant` (e.g. `brew install ant` and `sudo apt-get install ant`).
+When running unit tests in Eclipse, many of the tests depend on TLA⁺ module and model files in the repository.
+These tests need to know where to look for these files.
+For this reason, you will often need to modify the test run configurations by adding `-Dbasedir=${workspace_loc:tlatools}` to the VM arguments (not the program arguments) in the Arguments tab of the run configurations menu.
+You should then be able to run & debug JUnit tests in the Eclipse UI.
 
-Make sure that you're using Java 11+. If you're using a lower version you'll see compilation errors.
+Note that running build or test operations with Ant can interfere with Eclipse's build system and cause Eclipse to report nonexistent compilation errors in the UI; right-clicking the project in the Workspace pane then selecting "Refresh" should make these go away.
```

Comment:
Eclipse packages ant and can run ant directly.  We should encourage users to run ant from inside Eclipse to avoid inconsistencies.